src/node/txreconciliation.h

```diff
@@ -0,0 +1,82 @@
+ * 0.  Reconciliation protocol handshake.
+ * 1.  Once we receive a new transaction, add it to the set instead of announcing immediately.
+ * 2.  At regular intervals, a reconciliation initiator requests a sketch from the peer, where a
+ *     sketch is a compressed representation of short form IDs of the transactions in their set.
+ * 3.  Once the initiator received a sketch from the peer, the initiator computes a local sketch,
+ *     and combines the two sketches to attempt finding the difference in *sets*.
+ * 4a. If the difference was not larger than estimated, see SUCCESS below.
+ * 4b. If the difference was larger than estimated, initial reconciliation fails. The initiator
+ *     requests a larger sketch via an extension round (allowed only once).
+ *     - If extension succeeds (a larger sketch is sufficient), see SUCCESS below.
+ *     - If extension fails (a larger sketch is insufficient), see FAILURE below.
+ *
+ * SUCCESS. The initiator knows full symmetrical difference and can request what the initiator is
+ *          missing and announce to the peer what the peer is missing.
+ *
+ * FAILURE. The initiator notifies the peer about the failure and announces all transactions from
+ *          the corresponding set. Once the peer received the failure notification, the peer
+ *          announces all transactions from their set.
+
+ * This is a modification of the Erlay protocol (https://arxiv.org/abs/1905.10518) with two
+ * changes (sketch extensions instead of bisections, and an extra INV exchange round), both
+ * are motivated in BIP-330.
+ */
+class TxReconciliationTracker
+{
+    class Impl;
+    const std::unique_ptr<Impl> m_impl;
+
+public:
+    explicit TxReconciliationTracker(uint32_t recon_version);
+    ~TxReconciliationTracker();
+
+    /**
+     * Step 0. Generates initial part of the state (salt) required to reconcile with the peer.
+     * The salt used for short ID computation required for reconciliation.
+     * The function returns the salt.
+     * A peer can't participate in future reconciliations without this call.
+     * This function must be called only once per peer.
+     */
+    uint64_t PreRegisterPeer(NodeId peer_id);
+
+    /**
+     * Step 0. Once the peer agreed to reconcile with us, generate the state required to track
+     * ongoing reconciliations. Should be called only after pre-registering the peer and only once.
+     * Returns:
+     * - true if the peer was registered
+     * - false if the peer violates the protocol
+     * - nullopt if nothing was done (e.g., we haven't pre-registered this peer)
+     */
+    std::optional<bool> RegisterPeer(NodeId peer_id, bool peer_inbound, bool recon_requestor,
+                                     bool recon_responder, uint32_t peer_recon_version, uint64_t remote_salt);
+
+    /**
+     * Attempts to forget reconciliation-related state of the peer (if we previously stored any).
+     * After this, we won't be able to reconcile with the peer.
```

Comment:
in 182d984136066fd4f09b3e1a1bbd9b50c6cb87be

Nice doxygen comments. Can you specify here whether it's safe/not to call `ForgetPeer` multiple times for the same peer?

Reply:
What makes this function different from others, for which we don't mention this?